packages/app/src/cli/models/extensions/specifications/payments_app_extension_schemas/redeemable_payments_app_extension_schema.ts

```diff
@@ -0,0 +1,55 @@
+import {BaseSchema} from '../../schemas.js'
+import {zod} from '@shopify/cli-kit/node/schema'
+
+export type RedeemablePaymentsAppExtensionConfigType = zod.infer<typeof RedeemablePaymentsAppExtensionSchema>
+
+const MAX_LABEL_SIZE = 50
+
+export const RedeemablePaymentsAppExtensionSchema = BaseSchema.extend({
+  targeting: zod.array(zod.object({target: zod.literal('payments.redeemable.render')})).length(1),
+  api_version: zod.string(),
+  payment_session_url: zod.string().url(),
+  refund_session_url: zod.string().url().optional(),
+  capture_session_url: zod.string().url().optional(),
+  void_session_url: zod.string().url().optional(),
+  balance_url: zod.string().url(),
+  merchant_label: zod.string().max(MAX_LABEL_SIZE),
+  buyer_label: zod.string().max(MAX_LABEL_SIZE).optional(),
+  buyer_label_translations: zod.array(zod.object({locale: zod.string(), label: zod.string()})).optional(),
+  supported_countries: zod.array(zod.string()),
+  supported_payment_methods: zod.array(zod.string()),
+  test_mode_available: zod.boolean(),
+  redeemable_type: zod.literal('gift_card'),
+  checkout_payment_method_fields: zod.array(zod.string()).optional(),
```

Comment:
We are missing the `ui_extension_registration_uuid` property; are we planning on adding it later?

Reply:
@wikiti part of a future work